src/search.cpp

```diff
@@ -491,10 +491,8 @@ void Thread::search() {
           && !Threads.stop
           && !Threads.stopOnPonderhit)
           {
-              const int F[] = { failedLow,
-                                bestValue - mainThread->previousScore };
-
-              int improvingFactor = std::max(246, std::min(832, 306 + 119 * F[0] - 6 * F[1]));
+              double fallingEval = (306 + 119 * failedLow + 6 * (mainThread->previousScore - bestValue)) / 581.0;
+              fallingEval        = std::max(246.0, std::min(832.0, fallingEval));
```

Comment:
that's not quite right replace 246.0 and 832.0 by 0.4749 and 1.606 respectively.